main.go

```diff
@@ -54,6 +59,30 @@ func mainE(ctx context.Context) error {
 		}
 		subcommands = append(subcommands, cmd)
 	}
+	{
+		c := kustomizepatch.Config{
+			Logger: logger,
+		}
+		cmd, err := kustomizepatch.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		subcommands = append(subcommands, cmd)
+
+		// Make kustomizepatch main command if konfigure is running in
+		// container as a kustomize plugin. Kustomize does not know how to call
+		// sub-commands. This is discovered by setting KONFIGURE_MODE:
+		// "kustomizepatch" environment variable.
+		if v := os.Getenv(konfigureModeEnvVar); v == "kustomizepatch" {
```

Comment:
kustomize does not understand sub-commands and expects main command to be a KRM function. This is why `KONFIGURE_MODE=kustomizepatch` makes a `kustomize` call default to `kustomize kustomizepatch` call.